snapcraft/cli/store.py

```diff
@@ -115,7 +129,7 @@ def push(snap_file, release):
             'After pushing, an attempt to release to {} '
             'will be made'.format(channel_list))
 
-    snapcraft.push(snap_file, channel_list)
+    snapcraft.push(snap_file, channel_list, only_metadata, force_metadata)
```

Comment:
instead of passing the logic over to push, can we handle the logic to call `update_metadata` from here? The printing with logger in that method should also be brought over here (it is a bit hard with some of the current logic we have but  most of this stuff should be easily movable.

Reply:
It can be moved, but conceptually "it's a push" (complete, or only metadata). 

Another drawback of moving it to here is the need to call the private '_get_data_from_snap_file' from the private '_store".

I could start to move stuff around and/or make it public if you insist (but probably it would be better to do that in a separate branch, which I can commit to do if we decide this path)